src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -4037,28 +4037,24 @@ void AuraEffect::HandleAuraModIncreaseEnergyPercent(AuraApplication const* aurAp
         return;
 
     Unit* target = aurApp->GetTarget();
-
     Powers powerType = Powers(GetMiscValue());
-    // do not check power type, we can always modify the maximum
-    // as the client will not see any difference
-    // also, placing conditions that may change during the aura duration
-    // inside effect handlers is not a good idea
-    //if (int32(powerType) != GetMiscValue())
-    //    return;
-
     UnitMods unitMod = UnitMods(UNIT_MOD_POWER_START + powerType);
-    float amount = float(GetAmount());
 
-    if (apply)
-    {
-        target->HandleStatModifier(unitMod, TOTAL_PCT, amount, apply);
-        target->ModifyPowerPct(powerType, amount, apply);
-    }
-    else
-    {
-        target->ModifyPowerPct(powerType, amount, apply);
-        target->HandleStatModifier(unitMod, TOTAL_PCT, amount, apply);
-    }
+    float pct = float(GetAmount());
+
+    // Calculate the change.
+    float change = target->GetMaxPower(powerType);
+    ApplyPercentModFloatVar(change, pct, apply);
```

Comment:
Note in 3.3.5 I deleted most uses of ApplyPercentModFloatVar due to it losing information about base amount, also not blizzlike (ie a -100 aura reduces amount to 1 instead of 0 because of division by zero issues)

Check c69a7d1223b2ce108a221e5cc70f76cd80ae6675 if you want to see my approach

Reply:
Also I think @joschiwald ruined it again - we shouldn't apply pct directly to increased max power ...

Reply:
Example: maxpower = 13.000; pct = 30; apply negative pct to it - 13.000 * -0.3 => 3.900 again some value we can't use.

Reply:
No

```c++
void ApplyPercentModFloatVar(float& var, float val, bool apply)
{
    var *= (apply ? (100.0f + val) / 100.0f : 100.0f / (100.0f + val));
}
```

var = 13000, val = -30, apply = true ->
var = 13000 * (100 - 30) / 100 = 13000 * 0.7 = 9100

Reply:
continue: 9100 - 13000 to get the change = 3900
you lose mana buff, your max mana falls from 13.000 to 10.000, but u lose 3900 mana, not 3000..... :/

Reply:
... or in another way, would you be satisfied with 9100 / 10000 mana after the effect ends?

Reply:
@Mihapro: i have no idea what are you trying to calculate here, just see my example https://ideone.com/vhcnmS

Reply:
Your last calculation is exactly what I am trying to explain... you got 9100 mana instead of 10000 when doing (13000, -30.f)

Reply:
and that is correct in apply case of -30%, 13000 is your base and not 10000